stardis/plasma/molecules.py

```diff
@@ -0,0 +1,422 @@
+            ).to(1)
+        )
+
+        molecule_densities_div_partition_function = molecule_number_density.copy().div(
+            molecule_partition_function
+        )
+        molecule_densities_div_partition_function.index.name = "molecule"
+
+        # grab densities for n_lower - need to use linelist as the index and normalize by dividing by the partition function
+        linelist_with_density_div_partition_function = linelist.merge(
+            molecule_densities_div_partition_function,
+            how="left",
+            on=["molecule"],
+        )
+
+        n_lower = (
+            (
+                exponent_by_point
+                * linelist_with_density_div_partition_function[np.arange(points)]
+            ).values.T  # arange mask of the dataframe returns the set of densities of the appropriate ion for the line at each point
+            * linelist_with_density_div_partition_function.g_lo.values
+        )
+
+        linelist["f_lu"] = (
+            10**linelist.log_gf / linelist.g_lo
+        )  # vald log gf is "oscillator strength f times the statistical weight g of the parent level"  see 1995A&AS..112..525P, section 2. Structure of VALD
+
+        line_nus = (linelist.wavelength.values * u.AA).to(
+            u.Hz, equivalencies=u.spectral()
+        )
+
+        emission_correction = 1 - np.exp(
+            (
+                -const.h
+                / const.k_B
+                * np.outer(
+                    line_nus,
+                    1 / (t_electrons * u.K),
+                )
+            ).to(1)
+        )
+
+        alphas = pd.DataFrame(
+            (
+                ALPHA_COEFFICIENT
+                * n_lower
+                * linelist.f_lu.values
+                * emission_correction.T
+            ).T
+        )
+
+        if np.any(np.isnan(alphas)) or np.any(np.isinf(np.abs(alphas))):
+            raise ValueError(
+                "Some alpha_line from vald are nan, inf, -inf " " Something went wrong!"
+            )
+
+        alphas["nu"] = line_nus.value
+        linelist["nu"] = line_nus.value
+
+        # Linelist preparation below is taken from opacities_solvers/base/calc_alpha_line_at_nu
+        linelist["level_energy_lower"] = ((linelist["e_low"].values * u.eV).cgs).value
+        linelist["level_energy_upper"] = ((linelist["e_up"].values * u.eV).cgs).value
+
+        # Radiation broadening parameter is approximated as the einstein A coefficient. Vald parameters are in log scale.
+        linelist["A_ul"] = 10 ** (linelist["rad"]) / (
+            4 * np.pi
+        )  # see 1995A&AS..112..525P for appropriate units - may be off by a factor of 4pi
+
+        return alphas, linelist
+
+
+class AlphaLineShortlistValdMolecule(ProcessingPlasmaProperty):
+    """
+    Calculates the alpha values for each molecular line from Vald at each depth point. This is adapted from the AlphaLineShortlistVald calculation.
```

Comment:
Does `alpha` have a physical meaning, or is it just a coefficient?

Reply:
Alpha is the letter Rybicki and Lightman use for the absorption coefficient in cm^-1. Probably I can mention that explicitly in the docstrings though.